web-animations-2/Overview.bs

```diff
@@ -2353,6 +2376,152 @@ Items sorted earlier are executed before those sorted later.
+Like animations, [=animation triggers=] are associated with a [=timeline=]
+and are attached to an [=animation attachment range|attachment range=].
+
+Issue: Should there be any effect of triggers on scroll-driven animations?
+
+### Animation Trigger Active Interval ### {#trigger-active-interval}
+
+[=Animation triggers=] define an <dfn for="animation trigger">active interval</dfn>,
+which is the period of time during which the trigger is scheduled to produce its effect
+on its associated [=animation=]’s playback.
+Each [=animation trigger=] has only one such interval,
+which is defined by its [=default range=] and [=exit range=].
+
+### Animation Trigger State ### {#trigger-state}
+
+An [=animation trigger=] |trigger| with [=animation trigger type=] |type|
+and an associated [=animation=] |animation|
+has an internal <dfn export lt="animation trigger state">state</dfn> that can be one of the following:
+
+<dl dfn-for="animation trigger state" dfn-type=value>
+  <dt><dfn>idle</dfn>
+  <dd>
+    The [=animation effect=] associated |animation| remains in
+    its [=animation effect/before phase=] and stays at zero [=animation/start time=].
+
+  <dt><dfn>primary</dfn>
+  <dd>
+    The ''primary'' effect defined by |type| is applied to |animation|.
+
+  <dt><dfn>inverse</dfn>
+  <dd>
+    The ''inverse'' effect defined by |type| is applied to |animation|.
+
+</dl>
+
+Issue: Do we need a formal resolution on the spec of the idle state?
+
+The [=animation trigger state=] |state| of an [=animation trigger=] |trigger|
+is controlled by the [=animation attachment range|attachment range=] it is attached to,
+given a flag |did trigger|, as follows:
+
+<dl class=switch>
+  : If |trigger|’s [=local time=] is [=unresolved=],
+  ::
+    Then |state| is ''animation trigger state/idle'', and |did trigger| is set to false.
+
+  : Otherwise,
+  ::
+
+    <dl class=switch>
+      : If |trigger| is inside its [=animation trigger/active interval|active interval=],
+      ::
+        Then |state| is ''animation trigger state/primary'', and |did trigger| is set to true.
+
+      : Otherwise,
+      ::
+
+        <dl class=switch>
+          : If |did trigger| flag is false,
+          ::
+            Then |state| is ''animation trigger state/idle''.
+
+          : Otherwise,
+          ::
+            |state| is ''animation trigger state/inverse''.
+
+        </dl>
+
+    </dl>
+
+</dl>
+
+Issue: Should it be the correct/expected behavior for the "did trigger" flag to reset to false if the timeline becomes idle again?
+
+### Animation Trigger Types ### {#trigger-types}
+
+The behavior an [=animation trigger=] applies to its associated [=animation=]’s
+playback is defined by its <dfn export>animation trigger type</dfn>,
+which can be one of the following:
+
+<dl dfn-type=value dfn-for='animation trigger type'>
+  <dt><dfn>once</dfn>
+  <dd>
+    The ''primary'' behavior is [=play an animation|triggering=] the associated animation.
```

Comment:
I think this needs to define how the behavior is only applied once. E.g. the suggested triggered state associated with an animation, initially false

Reply:
I think this is covered in the new state updating algo, right?

Reply:
I suppose if the type is `once` and `did trigger` is `true` it should suffice to indicate that the state should not change, right?